core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -110,7 +110,14 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging {
 
     def abortQueuedWorkflows(submissionId: UUID) = {
       datasource.inTransaction { dataAccess =>
-        dataAccess.workflowQuery.batchUpdateWorkflowsOfStatus(submissionId, WorkflowStatuses.Queued, WorkflowStatuses.Aborted)
+        // Update workflows in batches of 10 to limit database deadlocks between
+        // this actor and WorkflowSubmissionActor. See GAWB-2243.
+        dataAccess.workflowQuery.listWorkflowRecsForSubmissionAndStatuses(submissionId, WorkflowStatuses.Queued).flatMap { workflows =>
+          val grouped = workflows.grouped(10).map { wfs =>
+            dataAccess.workflowQuery.batchUpdateStatus(wfs, WorkflowStatuses.Aborted)
+          }.toSeq
+          DBIO.fold(grouped, 0)(_ + _)
+        }
```

Comment:
Ooh, I might have an explanation. Before, the deadlock was between

transaction 1:
```
update WORKFLOW set status = 'Aborted', ... where status = 'Queued' and submission_id = ...
```
transaction 2:
```
update WORKFLOW set status = 'Launching', ... where id = ...
```
So T1 was doing a full scan on the `idx_workflow_status` index, locking any rows it finds. Then T2 comes in and locks the primary key index for a row already processed by the scan in T1, and boom, deadlock.

Now, the new query is a little different. T1 now looks like:
```
select id from WORKFLOW where status = 'Queued' and submissionId = ...
update WORKFLOW set status = 'Aborted' where id in (<ids returned from query above>)
```
The select is no longer locking the `idx_workflow_status` index, so there is no more deadlock (it would be locked if it were a "select for update"). So it's not really the batching that is fixing it, it's the different query structure. To test this, I re-ran my test with batchSize = Integer.MAX_VALUE (effectively disabling batching), and the deadlock still does not occur.

References:
https://dev.mysql.com/doc/refman/5.7/en/innodb-deadlocks.html
https://dev.mysql.com/doc/refman/5.7/en/innodb-deadlocks-handling.html
https://www.xaprb.com/blog/2006/08/03/a-little-known-way-to-cause-a-database-deadlock/

I'm still reading, not sure if the best solution yet, just posting my findings along the way.